### ISSUE TITLE
.NET Intrumentation - validate all env. vars. before start injecting env. vars.

### ISSUE THREAD

**Issue by Kielek** [open-telemetry/opentelemetry-operator#1094]
I think this behavior and the log message are misleading.

When `false` is returned then it interrupts processing next env vars.
It could leave the container in a "semi-instrumented" state. Is it expected?

This comment applies also to the code BEFORE the changes introduced in this PR. Therefore, it would be better to make a separate PR to address it.

_Originally posted by @pellared in https://github.com/open-telemetry/opentelemetry-operator/pull/1081#discussion_r969457713_

For now, .NET injecting process is validating env. variables and injecting it one by one, instead of make the whole validation upfront.

The goal is to validate first and inject env. variables only when needed.

**Comment by avadhut123pisal** [comment]
@Kielek  In case of .NET, during the validation phase, environment variables are set to the copy of container object and at last actual container object is updated. 

[	pod.Spec.Containers[index] = container](https://github.com/open-telemetry/opentelemetry-operator/blob/10404ebe147630a070768f346a541524672700ec/pkg/instrumentation/dotnet.go#L112)

In case of other language instrumentation (Java, Python, NodeJS), the problem described here do exists. 
I have raised the PR to fix that. Along with that, I have added a check for all language instrumentation, to skip the env var injection and OTEL SDK configuration, if the first step related to language SDK injection fails. 

Please let me know, if this is ok or some other cases needs to be handled.

**Comment by pellared** [comment]
@avadhut123pisal It is only a "shallow copy"  and the "original" container env vars can be modified here: https://github.com/open-telemetry/opentelemetry-operator/blob/10404ebe147630a070768f346a541524672700ec/pkg/instrumentation/dotnet.go#L133

**Comment by avadhut123pisal** [comment]
@pellared  I got your point. I will update the PR accordingly.

**Comment by Kielek** [comment]
@avadhut123pisal, I agree with @pellared, I have one more comment, in your draft PR.

**Comment by avadhut123pisal** [comment]
> * ALL validation should be done before processing. This means that  `container.Env[idx].ValueFrom != nil` for all env vars should be checked before setting anything.

@pellared  Looking at the existing implementation, it seems that we need to validate only those env vars, whose value needs to be concatenated with the value defined in Operator code, for eg. JAVA_TOOL_OPTIONS, NODE_OPTIONS. Other env vars can have their value populated using valueFrom syntax. 

@pellared  @Kielek  Please let me know your thoughts on this.
